From 6e1538828f2c68ce894e4590bfb91b478bc50a85 Mon Sep 17 00:00:00 2001
From: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Date: Fri, 23 Feb 2024 11:18:33 +0100
Subject: [PATCH 529/954] feat(stm32mp21): disable PWR_CPU3D3SR access on
 STM32MP21

PWR_CPU3D3SR is not present on STM32MP21.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 1a4e7e00f6ca3c01a6112ba4119a7280f57adeac feat(stm32mp21): disable PWR_CPU3D3SR access on STM32MP21]
Change-Id: I6ae61bed56bb308dd23d5711c081e19c8395f934
Signed-off-by: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/362445
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 990d3c63bc..812bcf8e16 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -670,12 +670,14 @@ static void __dead2 stm32_system_off(void)
 		mmio_setbits_32(rcc_base + RCC_C2RSTCSETR, RCC_C2RSTCSETR_C2RST);
 	}
 
+#if !STM32MP21
 	/* If CPU3 is not in reset */
 	if ((mmio_read_32(pwr_base + PWR_CPU3D3SR) & PWR_CPU3D3SR_CSTATE_MASK) != 0U) {
 		WARN("PSCI system off with Cortex M0 running.\n");
 		/* Force reset of CPU3 = Cortex M0+ */
 		mmio_setbits_32(rcc_base + RCC_C3CFGR, RCC_C3CFGR_C3RST);
 	}
+#endif /* !STM32MP21 */
 
 	/* Freeze all watchdog with shadow value of HCONF1 */
 	if (stm32_get_otp_index(HCONF1_OTP, &otp_idx, NULL) == 0U) {
-- 
2.53.0

